File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
class Squares(private val input: Int) {

    private fun Int.square() = this * this

    fun sumOfSquares() = (1..input).sumBy { it.square() }

    fun squareOfSum() = (1..input).sum().square()

    fun difference() = squareOfSum() - sumOfSquares()
}

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  attr_reader :range

  def initialize(input)
    @range = (1..input)
  end

  def square_of_sum
    range.sum.abs2
  end

  def sum_of_squares
    range.sum(&:abs2)
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
